tests/main/interfaces-libvirt/task.yaml

```diff
@@ -0,0 +1,82 @@
+    libvirt domains and other resources.
+
+    A snap which defines a libvirt plug must be shown in the interfaces list.
+    The plug must not be autoconnected on install and, as usual, must be able to be
+    reconnected.
+
+    A snap declaring a plug on this interface must be able to create and destroy a domain.
+    The test uses a snap that carries a unikernel built to be run on top of qemu, boot and
+    respond to ping. Once the domain is created, the test checks connectivity to the unikernel.
+
+prepare: |
+    echo "Given libvirt and qemu are installed"
+    sysctl -w net.ipv6.conf.all.disable_ipv6=1
+    trap "sysctl -w net.ipv6.conf.all.disable_ipv6=0" EXIT
+    apt install -y libvirt-bin qemu
+    adduser test libvirtd
+
+    echo "And libvirt is configured to manage /dev/net/tun"
+    systemctl stop libvirtd.service || true
+    echo 'cgroup_device_acl = ["/dev/net/tun", "/dev/random", "/dev/urandom"]' | tee -a /etc/libvirt/qemu.conf
+
+    echo "And the required services up"
+    systemctl start libvirtd.service
+    systemctl start virtlogd.socket
+
+    echo "And a snap declaring a plug on the libvirt interface is installed"
+    snap install --edge test-snapd-libvirt-consumer
+
+    echo "And the required tap interface is in place"
+    ip tuntap add tap100 mode tap
+    ip addr add 10.0.0.1/24 dev tap100
+    ip link set dev tap100 up
+
+restore: |
+    systemctl stop libvirtd.service
+    systemctl stop virtlogd.socket
+    apt remove -y --purge libvirt-bin qemu
+
+    ip link delete tap100
+
+execute: |
+    CONNECTED_PATTERN=":libvirt +test-snapd-libvirt-consumer"
+    DISCONNECTED_PATTERN="\- +test-snapd-libvirt-consumer:libvirt"
+
+    echo "The plug is not connected by default"
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "==================================="
+
+    echo "When the plug is connected"
+    snap connect test-snapd-libvirt-consumer:libvirt
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then the snap is able to create the unikernel domain"
+    su -l -c "test-snapd-libvirt-consumer.machine-up" test
+    virsh list | MATCH ping-unikernel
+
+    echo "And the unikernel is accesible"
+    ping -c 1 -q -W 1 10.0.0.2
+
+    echo "And the snap is able to destroy the unikernel domain"
+    su -l -c "test-snapd-libvirt-consumer.machine-down" test
+    virsh list | MATCH -v ping-unikernel
+
+    echo "==================================="
+
+    echo "When the plug is disconnected"
+    snap disconnect test-snapd-libvirt-consumer:libvirt
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "Then the snap is not able to create a domain"
+    if su -l -c "test-snapd-libvirt-consumer.machine-up 2>${PWD}/creation.error" test; then
+        echo "Expected permission error accessing libvirtd socket with disconnected plug"
+        exit 1
+    fi
+    grep -q "Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied" creation.error
```

Comment:
MATCH is a better approach here as well (cat file | MATCH). Otherwise, if it fails we have no idea about what was _actually_ in the file.